controller/service.go

```diff
@@ -35,7 +35,7 @@ const (
 	annotationIPAllocateFromPool = "metallb.universe.tf/ip-allocated-from-pool"
 )
 
-func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service) {
+func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service) bool {
```

Comment:
That's what I wanted to avoid, moving the logging outside because it won't be consistent with other logging in that file (not errors) and create an (in my view) unneeded "convergedFailed" error. We could return the controller sync state directly (controllers.SyncStateSuccess or controllers.SyncStateErrorNoRetry). It will more meaningful and simpler.

Reply:
Returning the controller enum would traverse too many levels of abstraction and I don't like it. 
The function errors, and it should return an error because that's what the reader expects.

The controllers.SyncState... are equally weird, but they were already there so we adopted them.

Reply:
Sorry, I was probably not clear. I'd prefer the function to return an error as it is what a function should do when an error condition happens.

Reply:
that's what I did

Reply:
oh, my bad